Expand on Neo4j and Query API (#634)
neo4j/docs-aura: modules/ROOT/pages/getting-started/connect-instance.adoc

| Field | Description

|Connection method

> |Neo4j is selected by default, and if you're learning about Neo4j you'll probably be alright with just leaving that selected.

Reviewer: ```suggestion
| The protocol is used for the communication between the Neo4j database server and the client application or tool.
The default is `neo4j+s//`.
See below for more information about connection methods.
```
We can't say that a user will _probably_ be alright.

|Password
|You are given the password when you initially create the instance
|===

== Connection method

> To connect to an instance, you can choose from the following connection methods: Neo4j or Query API.

Reviewer: ```suggestion
To connect to an instance, you select one of the following connection methods: Neo4j or Query API.
```

> *Neo4j*: The default is The default is Neo4j Bolt `neo4j+s//`.
> Connect through the Neo4j Bolt protocol using a WebSocket connection to the Neo4j port. 

Reviewer: ```suggestion
*`neo4j+s//`*: This is the default connection method. 
This scheme is built on  the `bolt://` protocol and uses a WebSocket connection to the Neo4j port. 
```

Use for best performance and stability.

*Query API*: Connect to Neo4j through the Query API over HTTPS on port 443 (a standard, widely open port). 
Use for debugging or when Bolt is unavailable due to network configuration.

> The connection method protocol is used for the communication between the Neo4j database server and the client application or tool.

Reviewer: ```suggestion

```
Move this back to the table.

For more information about connection schemes, see link:https://neo4j.com/docs/operations-manual/current/configuration/connectors/[Operations Manual -> Configure network connectors] and link:https://neo4j.com/docs/bolt/current/bolt/[Bolt Protocol].

== Errors 

In an enterprise environment with restrictive networking, connection errors are sometimes caused because non-standard ports used by the Bolt protocol are blocked by firewalls.

Neo4j tools have traditionally connected over Bolt, a binary protocol that offers optimal performance and stability, that requires an open WebSocket connection to the Neo4j port.

> 

Reviewer: ```suggestion
```
Is this true? I think `neo4j` has been around for quite some time and as mentioned above, it's built on bolt but it is not the same, so we should probably say that.

|Password
|You are given the password when you initially create the instance
|===

== Connection methods

> To connect to an instance, you select one of the following connection methods: Neo4j or Query API.

Reviewer: ```suggestion
To connect to an instance, you select one of the following connection methods:
```

*Neo4j*: `neo4j+s//` This is the default connection method. 
This scheme is built on the `bolt://` protocol and uses a WebSocket connection to the Neo4j port. 
Use for best performance and stability.

> *Query API*: Connects to Aura through the Query API via HTTPS on port 443 (a standard, widely open port, ensuring compatibility with most firewalls). 

Reviewer: ```suggestion
*Query API*: Connects to Aura through the Query API via HTTPS on port 443 (a standard, widely open port, compatible with most firewalls). 
```

> Use for debugging or when Bolt is unavailable due to your organization's network configuration.

Reviewer: ```suggestion
Use for debugging or when Bolt is unavailable due to network configurations.
```

For more information about connection schemes, see link:https://neo4j.com/docs/operations-manual/current/configuration/connectors/[Operations Manual -> Configure network connectors] and link:https://neo4j.com/docs/bolt/current/bolt/[Bolt Protocol].

> == Troubleshooting

Reviewer: ```suggestion
=== Connection error
```
_Troubleshooting_ indicates a list of problems and their solutions, this is only one problem so it's better to just call it what it is. Also, it doesn't need a level 2 header.

> If you encounter connection errors with the Neo4j connection method, try switching to Query API. 

Reviewer: ```suggestion
If you encounter a connection error with the `neo4j://` connection method, try switching to Query API. 
```

In an enterprise environment with restrictive networking, connection errors are sometimes caused because non-standard ports used by the Neo4j protocol are blocked by firewalls.

> == Query API for Aura

Reviewer: I don't think this is the spot for this either. We need to find a better place for it. Let's talk about it!

The Query API allows interaction with an Aura database using Cypher via HTTPS.

For more information, see the link:https://neo4j.com/docs/query-api/current/[Query API documentation] but note that:

* for Aura instances host is of the format `<databaseID>.databases.neo4j.io`

> * the on-prem documentation refers to port `:7474` and this is true for a on-prem install but Aura, given its HTTPS, the <port> defaults as 443.

Reviewer: What do you mean with "on-prem"? This is not terminology we use externally.
